merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -503,6 +505,14 @@ public SimulationResults computeResults(
                                  serializedTimeline);
   }
 
+  @SuppressWarnings("unchecked")
+  private static <DirectiveReturn, TaskReturn> SerializedValue serializeReturnValue(
+      final Directive<?, ?, DirectiveReturn> directive,
+      final TaskReturn returnValue) {
+    // SAFETY: Tasks always return the same type as the TaskSpecType that declares them
+    return directive.directiveType().serializeReturnValue((DirectiveReturn) returnValue);
```

Comment:
I feel like there must be some way to make `DirectiveReturn` and `TaskReturn` manifestly the same type, i.e. make this method take one type parameter. 🤔 I might look at this more carefully later today.

Reply:
Future work